lib/prawn/embedded_files/filespec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Prawn
+  module EmbeddedFiles
+    class Filespec
+      attr_reader :file_name
+
+      def initialize(file, options = {})
+        name = options[:name] || file.checksum
+
+        @file_name = PDF::Core::LiteralString.new(name)
+
+        if options[:description]
+          desc_str = options[:description].to_s
+          @description = PDF::Core::LiteralString.new(desc_str)
+        end
+
+        @hidden = options[:hidden]
+        @file = file
+      end
+
+      def hidden?
+        @hidden
+      end
+
+      def build_pdf_object(document)
+        obj = document.ref!(
+          Type: :Filespec,
+          F: file_name,
+          EF: { F: file },
```

Comment:
Since the file specification dictionary contains the /F and /UF entries, it would makes sense to also use the /UF entry inside the /EF key, i.e. `EF: { F: file, UF: file }`.

I don't think this is required but it doesn't add much in terms of PDF size and we would be on the safe side if some PDF readers expect the key to be there.

Reply:
Sure, probably it's better to use it